packages/swing-store/src/bundleStore.js

```diff
@@ -149,9 +166,20 @@ export function makeBundleStore(db, ensureTxn, noteExport = () => {}) {
     // prettier-ignore
     (parts.length === 2 && type === 'bundle') ||
       Fail`expected artifact name of the form 'bundle.{bundleID}', saw ${q(name)}`;
-    const bundle = getBundle(bundleID);
-    bundle || Fail`bundle ${q(name)} not available`;
-    yield* Readable.from(Buffer.from(decodeBase64(bundle.endoZipBase64)));
+    if (bundleID.startsWith('b0-')) {
+      // artifact is the raw JSON-serialized bundle data
+      const rawBundle = sqlGetBundle.get(bundleID);
+      rawBundle || Fail`bundle ${q(name)} not available`;
+      yield* Readable.from(Buffer.from(rawBundle));
+    } else if (bundleID.startsWith('b1-')) {
+      const bundle = getBundle(bundleID);
```

Comment:
Line 171 is doing a raw SQL query, whereas line 175 is calling `getBundle`.  This should use `getBundle` for either case (if that doesn't work we have a problem in `getBundle`).

Reply:
`getBundle` returns the parsed bundle object, but the canonical (hashed) format of a `b0-` bundle is a JSON-serialized string. Using `getBundle` for `b0-` artifacts would have required:
```js
const bundle = getBundle(bundleID); // yields object
const artifact = JSON.stringify(bundle); // back to string
yield* Readable.from(Buffer.from(artifact));
```

which would expand to `yield* Readable.from(Buffer.from(JSON.stringify(JSON.parse(sqlGetBundle.get(bundleID)))))`, and the redundant JSON parse+stringify felt wrong.

You're right that it indicates a deeper problem. For `b1-` format bundles, `getBundle()` reads a blob, decompresses it, base64-encodes it, and wraps it in an object. Then `exportBundle` undoes the base64 encoding.

I think both paths want to use something other than getBundle. The `b0-` format wants to use the DB result directly. The `b1-` format needs to decompress that result, but doesn't want any other formatting to be applied. That means the decompression logic needs to be duplicated, or factored out.

I didn't want to change too much about the way `b1-` bundles are handled, so I left the redundancy present on that path, using a different approach for the `b0-` path to avoid starting off with a redundancy.

Reply:
huh, ok so the cleanest path is to just return `yield* Readable.from(Buffer.from(rawBundle))` in both cases. If we're using `b0-` then the DB is holding the JSON-stringified bundle object (which is also the artifact), and if we're using `b1-` then the DB is holding the raw bytes of the zip file (which is also the artifact).